### PR TITLE
Normalise matrix naming

### LINKED ISSUE #8492: Rename View matrices in the form world_to_view, etc
Currently, the View uniform looks like this:
```rust
struct View {
    view_proj: mat4x4<f32>,
    unjittered_view_proj: mat4x4<f32>,
    inverse_view_proj: mat4x4<f32>,
    view: mat4x4<f32>,
    inverse_view: mat4x4<f32>,
    projection: mat4x4<f32>,
    inverse_projection: mat4x4<f32>,
    ...
};
```

There's been much confusion over which of these do what. It would be much better to call, for instance the projection matrix, view_to_clip. We could then put "This is the projection matrix" in a doc comment.

We could also use a doc comment for unjittered_view_proj vs view_proj. The unjittered matrix is basically only useful for calculating motion vectors for TAA, and is equivalent to the view_proj matrix when the motion vector prepass is disabled. Users shouldn't really need to touch the unjittered matrix ever.

### LINKED ISSUE #10909: View matrix and Inverse View matrix mixed up.
## Bevy version

12.0

## Problem

The naming of the `view` and `inverse_view` matrices in the `View` struct are the opposite of what those names mean everywhere else.

A view matrix transforms from world to view space. The "view matrix" in the `View` struct transforms from view to world space.

https://github.com/bevyengine/bevy/blob/a4c27288c773959ff307635b4ac25399e96f5de7/crates/bevy_render/src/view/mod.rs#L381

### PR DESCRIPTION
**Pull request by ricky26**
# Objective
- Fixes #10909
- Fixes #8492

## Solution
- Name all matrices `x_from_y`, for example `world_from_view`.

## Testing
- I've tested most of the 3D examples. The `lighting` example particularly should hit a lot of the changes and appears to run fine.

---

## Changelog
- Renamed matrices across the engine to follow a `y_from_x` naming, making the space conversion more obvious.

## Migration Guide
- `Frustum`'s `from_view_projection`, `from_view_projection_custom_far` and `from_view_projection_no_far` were renamed to `from_clip_from_world`, `from_clip_from_world_custom_far` and `from_clip_from_world_no_far`.
- `ComputedCameraValues::projection_matrix` was renamed to `clip_from_view`.
- `CameraProjection::get_projection_matrix` was renamed to `get_clip_from_view` (this affects implementations on `Projection`, `PerspectiveProjection` and `OrthographicProjection`).
- `ViewRangefinder3d::from_view_matrix` was renamed to `from_world_from_view`.
- `PreviousViewData`'s members were renamed to `view_from_world` and `clip_from_world`.
- `ExtractedView`'s `projection`, `transform` and `view_projection` were renamed to `clip_from_view`, `world_from_view` and `clip_from_world`.
- `ViewUniform`'s `view_proj`, `unjittered_view_proj`, `inverse_view_proj`, `view`, `inverse_view`, `projection` and `inverse_projection` were renamed to `clip_from_world`, `unjittered_clip_from_world`, `world_from_clip`, `world_from_view`, `view_from_world`, `clip_from_view` and `view_from_clip`.
- `GpuDirectionalCascade::view_projection` was renamed to `clip_from_world`.
- `MeshTransforms`' `transform` and `previous_transform` were renamed to `world_from_local` and `previous_world_from_local`.
- `MeshUniform`'s `transform`, `previous_transform`, `inverse_transpose_model_a` and `inverse_transpose_model_b` were renamed to `world_from_local`, `previous_world_from_local`, `local_from_world_transpose_a` and `local_from_world_transpose_b` (the `Mesh` type in WGSL mirrors this, however `transform` and `previous_transform` were named `model` and `previous_model`).
- `Mesh2dTransforms::transform` was renamed to `world_from_local`.
- `Mesh2dUniform`'s `transform`, `inverse_transpose_model_a` and `inverse_transpose_model_b` were renamed to `world_from_local`, `local_from_world_transpose_a` and `local_from_world_transpose_b` (the `Mesh2d` type in WGSL mirrors this).
- In WGSL, in `bevy_pbr::mesh_functions`, `get_model_matrix` and `get_previous_model_matrix` were renamed to `get_world_from_local` and `get_previous_world_from_local`.
- In WGSL, `bevy_sprite::mesh2d_functions::get_model_matrix` was renamed to `get_world_from_local`.

### CODE REVIEW

File: assets/shaders/irradiance_volume_voxel_visualization.wgsl
Comment (superdump) [inline assets/shaders/irradiance_volume_voxel_visualization.wgsl:7]: Hmm. Why? ‘Voxel’ isn’t a space?

File: crates/bevy_pbr/src/render/light.rs
Comment (superdump) [inline crates/bevy_pbr/src/render/light.rs:656]: I feel like this is maybe world from view.

File: crates/bevy_pbr/src/render/light.rs
Comment (superdump) [inline crates/bevy_pbr/src/render/light.rs:1113]: This gets assigned to world from view and I think the spot light view matrix function creates a view matrix in bevy’s current terms which means world from view.

File: crates/bevy_pbr/src/light/mod.rs
Comment (superdump) [inline crates/bevy_pbr/src/light/mod.rs:1829]: As noted, this function should be called world from view and so should the variable. I noticed this because the calculation below didn’t make sense with the new naming.

File: crates/bevy_pbr/src/light/mod.rs
Comment (superdump) [inline crates/bevy_pbr/src/light/mod.rs:1831]: This should then be clip from world = clip from view * world from view inverse. World from view inverse meaning the same as view from world.

File: examples/3d/irradiance_volumes.rs
Comment (superdump) [inline examples/3d/irradiance_volumes.rs:136]: Same here with voice not being a space, but now I have seen the light from world and so on and I like that. I’m open for debate.

**Review by superdump — CHANGES_REQUESTED** [review]
Mostly looks great, thanks for doing it. Left a couple of comments.

File: assets/shaders/irradiance_volume_voxel_visualization.wgsl
Comment (ricky26) [reply assets/shaders/irradiance_volume_voxel_visualization.wgsl:7]: I'm not particular about this change, but it made sense as a shorthand to me as the space in which the irradiance voxels are 1 unit in size and aligned to the origin.

Are you objecting to its name / lack of documentation, or the fact that I considered the world -> irradiance voxel matrix as a mapping between spaces?

File: crates/bevy_pbr/src/render/light.rs
Comment (ricky26) [reply crates/bevy_pbr/src/render/light.rs:656]: Yep, it's being built from `GlobalTransform` too, which should've been a hint. :sweat_smile:

**Comment by ricky26** [comment]
I've applied some review feedback, merged latest main back in and tested several examples (particularly `lighting`, which should test most of the changes).

**Comment by IceSentry** [comment]
@ricky26 can you fix the conflicts?

**Comment by ricky26** [comment]
I've merged main in again after the skybox motion vectors and fixed up the introduced changes.

File: assets/shaders/irradiance_volume_voxel_visualization.wgsl
Comment (superdump) [reply assets/shaders/irradiance_volume_voxel_visualization.wgsl:7]: I’ve changed my mind after seeing the light transforms.

File: examples/3d/irradiance_volumes.rs
Comment (superdump) [reply examples/3d/irradiance_volumes.rs:136]: I’m ok with this now.

**Review by superdump — APPROVED** [review]
Assuming this is well tested, LGTM.

**Comment by ricky26** [comment]
I think I've run enough examples to touch all of the changed code. I did catch some missed renames in the meshlet shaders.

I've finished filling in the migration notes with everything public that's been renamed.

**Review by IceSentry — APPROVED** [review]
I tested a whole bunch of examples and didn't find any regressions. Thank you for working on this!

Approving, assuming the conflicts are fixed of course.

**Comment by alice-i-cecile** [comment]
@ricky26 once merge conflicts are resolved holler at me and I'll merge this in.

**Comment by alice-i-cecile** [comment]
Thank you to everyone involved with the authoring or reviewing of this PR! This work is relatively important and needs release notes! Head over to https://github.com/bevyengine/bevy-website/issues/1331 if you'd like to help out.

**Pull request merged**
